components/style/data.rs

```diff
@@ -175,25 +175,24 @@ impl NodeData {
         self.styles = match mem::replace(&mut self.styles, Uninitialized) {
             Uninitialized => Previous(f()),
             Current(x) => Previous(Some(x)),
-            _ => panic!("Already have previous styles"),
```

Comment:
I guess this change is because we call `gather_previous_styles` unconditionally on all the children even if we end up not styling them, is that right?